### PR TITLE
Admin impersonation

### LINKED ISSUE #151: Admins should be able to impersonate anyone

### PR DESCRIPTION
**Pull request by kylesferrazza**
closes #151 

- [x] UI for admins to choose a user to impersonate
- [x] show impersonating and stop button in navbar

### CODE REVIEW

**Comment by kylesferrazza** [comment]
@blerner does professors impersonating any of their students seem right?


What if user A is a professor for a course that has a student B who is a professor of one of user A's courses?

**Comment by blerner** [comment]
Yes, that sounds fine.  Professors intrinsically have a higher level of trust, and a higher professional responsibility, so if there's ever a situation where A and B are mutually professors of each other, that has to get resolved out-of-band with respect to this server.  Yes, technically, A could impersonate B to impersonate A, and (maybe) that means that they'd have B's privileges to modify A's grades...but that's not an attack scenario I'm worried about.

**Comment by kylesferrazza** [comment]
Alright, this one is just for admins (opened #166), please review

File: app/graphql/mutations/stop_impersonating.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Mutations
+  class StopImpersonating < BaseMutation
+    field :success, Boolean, null: false
+
+    def authorized?
```
Comment (blerner) [inline app/graphql/mutations/stop_impersonating.rb:7]: I think this is probably overly strict: if `true_user != current_user`, then you should be able to stop impersonating.  Stopping impersonation always drops privileges, so it's always a conservative action to take...as long as we're careful that the only way `true_user` can become set is via `ImpersonateUser`, which in turn does the correct access checks.

File: app/graphql/types/query_type.rb [hunk 2]
```diff
@@ -43,5 +48,12 @@ def registration(id:)
     def course(id:)
       HourglassSchema.object_from_id(id, context)
     end
+
+    field :users, [UserType], null: false do
+      guard Guards::CURRENT_USER_ADMIN
```
Comment (blerner) [inline app/graphql/types/query_type.rb:53]: Subtle!  But correct -- in Bottlenose, an admin can only view all users when they're logged in as admin, and not currently  impersonating someone else

**Comment by kylesferrazza** [comment]
An admin logs in and just sees a list right now:
image

And then while impersonating, the navbar changes to look like this:

image

**Comment by blerner** [comment]
When an admin logs in, they immediately see that list of _all_ users?  Or is there a page for them to go to first, to choose whom to impersonate?  (In my case, I'm a professor and an admin, and most of the time I need to see my courses, not impersonation stuff...)

**Pull request merged**